src/main/java/Sinep/Sinep.java

```diff
@@ -0,0 +1,322 @@
+                if (!dataFile.getParentFile().exists()) {
+                    Files.createDirectories(dataFile.getParentFile().toPath());
+                }
+            } catch (IOException e) {
+                System.out.println("An error occurred while trying to create the file: " + e.getMessage());
+                return; // Exit the method if file creation fails
+            }
+        }
+        try {
+            Scanner scanner = new Scanner(taskFile);
+            int counter = 0;
+            while (scanner.hasNext()) {
+                String nextLine = scanner.nextLine();
+                if (nextLine.startsWith("[T]")) {
+                    String description = nextLine.substring(6).trim();
+                    taskList.add(new Todo(description));
+                    if (nextLine.charAt(4) == 'X') {
+                        Task markingTask = taskList.get(counter);
+                        markingTask.markAsDone();
+                    }
+                } else if (nextLine.startsWith("[D]")) {
+                    String command = getDeadlineCommand(nextLine);
+                    System.out.println(command);
+                    taskList.add(new Deadline(command));
+                    if (nextLine.charAt(4) == 'X') {
+                        Task markingTask = taskList.get(counter);
+                        markingTask.markAsDone();
+                    }
+                } else {
+                    String command = getEventCommand(nextLine);
+                    taskList.add(new Event(command));
+                    if (nextLine.charAt(4) == 'X') {
+                        Task markingTask = taskList.get(counter);
+                        markingTask.markAsDone();
+                    }
+                }
+                counter++;
+            }
+        } catch (FileNotFoundException e) {
+            File parentDir = taskFile.getParentFile();
+            if (!parentDir.exists() && parentDir.mkdirs()) {
+                throw new IOException("File not found.");
+            }
+        }
+    }
+
+    private static String getEventCommand(String nextLine) {
+        int startDescription = nextLine.indexOf("[E]") + 6;
+        int endDescription = nextLine.indexOf(" (from:");
+        String description = nextLine.substring(startDescription, endDescription).trim();
+        String fromKeyword = "(from: ";
+        String toKeyword = " to:";
+        int start = nextLine.indexOf(fromKeyword) + fromKeyword.length();
+        int end = nextLine.indexOf(toKeyword);
+        String fromTime = nextLine.substring(start, end).trim();
+        String startKeyword = "to: ";
+        int start1 = nextLine.indexOf(startKeyword) + startKeyword.length();
+        int end1 = nextLine.lastIndexOf(')');
+        String toTime = nextLine.substring(start1, end1).trim();
+        String command;
+        command = "event " + description + " /from " + fromTime + " /to " + toTime;
+        return command;
+    }
```

Comment:
It is advisable to have logically linked code to be put together, and other part separated by a line break. Please make these changes everywhere you see fit.

Example: (Add line breaks as you deem right)
```suggestion
    private static String getEventCommand(String nextLine) {
        int startDescription = nextLine.indexOf("[E]") + 6;
        int endDescription = nextLine.indexOf(" (from:");
        String description = nextLine.substring(startDescription, endDescription).trim();
        
        String fromKeyword = "(from: ";
        String toKeyword = " to:";
        int start = nextLine.indexOf(fromKeyword) + fromKeyword.length();
        int end = nextLine.indexOf(toKeyword);
        String fromTime = nextLine.substring(start, end).trim();
        
        String startKeyword = "to: ";
        int start1 = nextLine.indexOf(startKeyword) + startKeyword.length();
        int end1 = nextLine.lastIndexOf(')');
        String toTime = nextLine.substring(start1, end1).trim();
        
        String command;
        command = "event " + description + " /from " + fromTime + " /to " + toTime;
        
        return command;
    }
```

Reference: https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically